### PR TITLE
Add process for security and vulnerability reporting

### PR DESCRIPTION
**Pull request by fredjn**
### Applicable Issues
fixes: #61 

### Description of the Change
This change proposes a lightweight security and vulnerability reporting proces.

### Alternate Designs
N/A

### Benefits
The eiffel-community can use the process to receive and properly handle security and vulnerability issues.

### Possible Drawbacks
None

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Fredrik Fristedt <fredrik.fristedt@axis.com>

### CODE REVIEW

**Comment by fredjn** [comment]
Note that I have not yet created the proposed private security e-mail list.

File: SECURITY.md
Comment (fdegir) [inline SECURITY.md:1]: Should we also include response in the heading since the doc talks about how the responses are done?

File: SECURITY.md
Comment (fdegir) [inline SECURITY.md:4]: Would be good to have a link to a page where all the reported/fixed issues are listed so people can be encouraged to check this before submitting a new issue?

The page could be linked to eiffel-community.github.io; something like https://eiffel-community.github.io/security.html

File: SECURITY.md
Comment (fdegir) [inline SECURITY.md:20]: Should we link to the page where all the reported/fixed issues are listed e.g. https://eiffel-community.github.io/security.html?

File: SECURITY.md
Comment (fdegir) [inline SECURITY.md:10]: If the project maintainers/contributors identify such issues themselves, how they should handle it? (the process taking the shortcut we talked about during the TC meeting)

They could perhaps still notify the security officer/community security volunteers once the issue is fixed so it can be listed on the security page together with other issues.

**Review by fdegir — COMMENTED** [review]
comments inline

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md:10]: > If the project maintainers/contributors identify such issues themselves, how they should handle it? (the process taking the shortcut we talked about during the TC meeting)
> 
> They could perhaps still notify the security officer/community security volunteers once the issue is fixed so it can be listed on the security page together with other issues.

I did not fully understand why the maintainers would have the privilege to shortcut the process, and also since the process is light weight it shouldn't really be a problem to follow it. I'm actually leaning towards "same process for all".

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Fixed

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Fixed

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Fixed

File: SECURITY.md
Comment (fdegir) [reply SECURITY.md:10]: This was based on the discussion we had during the meeting on Monday. But I agree with you, they should follow the process.

But, I still think this, _projects themselves following the process_, should somehow be stated so we don't miss the issues found & fixed by the projects themselves. Otherwise, those issues will be buried deep in the history, mixed with the regular development activities, making them invisible to the users who may be using vulnerable version.

Comment (fredjn) [reply SECURITY.md:10]: Agreed, I'll add something to clarify.

File: SECURITY.md
Comment (m-linner-ericsson) [inline SECURITY.md:4]: Sorry if I have missed something, please correct me. Shouldn't we link to the pull request to create the github.io page in this request?

**Review by m-linner-ericsson — CHANGES_REQUESTED** [review]

File: SECURITY.md
Comment (fdegir) [reply SECURITY.md:4]: i think that needs to be done once the process is agreed on.

Comment (fredjn) [reply SECURITY.md:4]: There is a similar sentence under the public disclosure paragraph. I don't think we need to have it both places. @eiffel-community/eiffel-community-maintainers while you are eagerly reviewing this ;), think about were do we want the link?

File: SECURITY.md
Comment (m-linner-ericsson) [inline SECURITY.md:4]:
```suggestion
We as a community encourage researchers, users and contributors to report vulnerabilities and security related issues to the eiffel community. All issues are thoroughly investigated by a community security officer and/or other community security volunteers. All reported and fixed security and vulnerability issues can be found on the [eiffel community security page](https://eiffel-community.github.io/security.html) **To be created**.
```

**Review by m-linner-ericsson — CHANGES_REQUESTED** [review]
Please ignore this if you think it is unnecessary I didn't like the idea of having a dead link in a security document as I interpreted @fdegir to vote for merging this and then creating the github.io page.

File: SECURITY.md
Comment (m-linner-ericsson) [reply SECURITY.md:4]: I posted a change suggestion on how to solve this.

**Comment by fredjn** [comment]
I removed the reminders/pace holders. If somebody hasn't done it already, I'll file an issue to create the page on eiffel-community.github.io

**Comment by fredjn** [comment]
https://github.com/eiffel-community/eiffel-community.github.io/issues/14

File: SECURITY.md
Comment (d-stahl-ericsson) [inline SECURITY.md:20]:
```suggestion
The eiffel-community humbly asks all vulnerability reporters to hold off on public disclosure and instead negotiate a time frame within which the vulnerability report will be processed, fixed and released by the eiffel-community. Once released it will be listed on the [eiffel community security page](https://eiffel-community.github.io/security.html) .
```

Comment (d-stahl-ericsson) [inline SECURITY.md:20]:
```suggestion
The eiffel-community humbly asks all vulnerability reporters to hold of on public disclosure and instead negotiate a time frame within which the vulnerability report will be processed, fixed and released by the eiffel-community. Once released it will be listed on the [Eiffel community security page](https://eiffel-community.github.io/security.html) .
```

File: SECURITY.md
Comment (d-stahl-ericsson) [inline SECURITY.md:4]:
```suggestion
We as a community encourage researchers, users and contributors to report vulnerabilities and security related issues to the Eiffel community. All issues are thoroughly investigated by a community security officer and/or other community security volunteers. All reported and fixed security and vulnerability issues can be found on the [eiffel community security page](https://eiffel-community.github.io/security.html) .
```

**Review by d-stahl-ericsson — CHANGES_REQUESTED** [review]
I'm happy with this; I only have a few suggestions on spelling and capitalization.

**Comment by m-linner-ericsson** [comment]
@fredjn My idea with the reminder was to have them and only remove them after the page has been created. Thus avoiding us posting a document with dead links. I guess you had some other idea that I missed.

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Github wouldn't let me commit the suggestions using the convinence-button, added your updates in a separate commit.

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Github wouldn't let me commit the suggestions using the convinence-button, added your updates in a separate commit.

**Comment by fredjn** [comment]
https://github.com/eiffel-community/eiffel-community.github.io/pull/15

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md:10]: Added some clarification on this.

**Comment by fredjn** [comment]
Ping! @eiffel-community/eiffel-community-maintainers

**Review by m-linner-ericsson — APPROVED** [review]

**Review by sodero — APPROVED** [review]

**Review by k-hallen-ericsson — APPROVED** [review]

**Review by fdegir — APPROVED** [review]

File: SECURITY.md
Comment (henning-roos) [inline SECURITY.md:6]: Should probably be Title Case in the header. I.E. "How to Report a Security Vulnerability"

**Review by henning-roos — CHANGES_REQUESTED** [review]
Looks good! I only had a comment to fix before I will approve the PR

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: fixed

**Review by henning-roos — APPROVED** [review]

**Review by d-stahl-ericsson — APPROVED** [review]

File: SECURITY.md
Comment (e-backmark-ericsson) [inline SECURITY.md:7]: Please rephrase to 'from a project within the Eiffel community or from an external contributor.'

File: SECURITY.md
Comment (e-backmark-ericsson) [inline SECURITY.md:9]: Before merging this PR I assume we need to appoint the 'security officer' and maybe also a group of 'security volunteers' within the community, to be able to live up to this promise.

**Review by e-backmark-ericsson — CHANGES_REQUESTED** [review]
The security email list should be in place before merging this PR

File: SECURITY.md
Comment (fredjn) [reply SECURITY.md]: Done

**Review by sodero — APPROVED** [review]

File: SECURITY.md
Comment (fdegir) [reply SECURITY.md:9]: I think this comment is addressed during the last TC meeting.

I have an action item from the meeting to update the governance document and write the term of security contacts.
I can add Fredrik and Kristofer in that section as well instead of adding to this doc and include link to that section in this doc so the process remains same but the names are handled in governance document?

Comment (e-backmark-ericsson) [reply SECURITY.md:9]: Yes, of course you could link to the governance docs for some of the details in the process. I'm not sure we should/need to disclose any names of the security officers/volunteers in a public document though. Should we? If not, where should we keep such information? What is the best practice in this area? Do most projects disclose their security officers/volunteers publicly?
And for the security mail list I assume this PR depends on #60 ?

Comment (fdegir) [reply SECURITY.md:9]: I've seen both - some communities disclose the names of the team members while others just call "Security Team" without listing people. Putting communities that list individuals as examples.

https://security.openstack.org
https://wiki.jenkins.io/display/JENKINS/Team+Leads

Comment (e-backmark-ericsson) [reply SECURITY.md:9]: Ok. The Jenkins info there is very old, and I think that OpenStack plays in a different league than Eiffel does. I'd vote for that we don't need to keep a public list of security representatives within the community, but for sure we need to have them noted in some internal TC document. But where in that case? HackMD? Google Docs?

Comment (fredjn) [reply SECURITY.md:9]: So, should I link something from the process document or not?

Comment (e-backmark-ericsson) [reply SECURITY.md:9]: I'd say that the governance document that Fatih is about to propose an update to should include some description of what the 'security officer' and 'security volunteer' roles are, and then you should be able to link to those from this document.
And of course the people behind those roles need to be appointed and added to relevant mail lists (?) before this PR is merged, so we can live up to the promises made here. I suggest that we appoint those people on the next TC meeting.

Comment (fdegir) [reply SECURITY.md:9]: We appointed "security officers" - Fredrik and Kristofer. I assume Kristofer made Fredrik a member of the maillist/group he created as well so this PR does not seem to have anything left to be fixed.

As part of the AP I got, I'll update the governance document and add the role "Security Officers" with the responsibilities and terms which is tied to TC term. For that, I need this doc to be merged. To close the gap, I'll add a link to that section from governance document to this document as well which I'll do that within the PR.

I think we should merge this PR and continue the conversation on listing the individuals volunteered/appointed as security officers under the new PR.

Comment (k-hallen-ericsson) [reply SECURITY.md:9]: Yes, we both me and Fredrik are on the list. Ok to merge.

Comment (e-backmark-ericsson) [reply SECURITY.md:9]: Ok. Sounds fine.

**Review by e-backmark-ericsson — APPROVED** [review]

**Comment by fredjn** [comment]
Need one more approval on eiffel-community/eiffel-community.github.io#15 before merge.

**Pull request merged**
